packages/flutter/lib/src/widgets/scroll_position_with_single_context.dart

```diff
@@ -0,0 +1,417 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/physics.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:meta/meta.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'gesture_detector.dart';
+import 'scroll_activity.dart';
+import 'scroll_context.dart';
+import 'scroll_notification.dart';
+import 'scroll_physics.dart';
+import 'scroll_position.dart';
+
+/// A scroll position that manages its own scroll activities.
+///
+/// This class combines the [ScrollPosition] logic (which implements
```

Comment:
these docs are out of date. search for ScrollIndependentPosition and ScrollPositionReadInterface in particular